### PR TITLE
Fix/dashboard restaurant screen

### PR DESCRIPTION
**Pull request by mujtaba912**

### CODE REVIEW

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/usecase/FilterRestaurantsUseCase.kt [hunk 1]
```diff
@@ -2,45 +2,17 @@ package org.thechance.common.domain.usecase
...
-import kotlin.math.floor
 
 interface IFilterRestaurantsUseCase {
-    suspend fun filterRestaurants(rating: Double, priceLevel: Int): List<Restaurant>
-    suspend fun searchFilterRestaurants(
-        restaurantName: String,
-        rating: Double,
-        priceLevel: Int
-    ): List<Restaurant>
+    suspend fun invoke(rating: Double, priceLevel: Int): List<Restaurant>
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/usecase/FilterRestaurantsUseCase.kt:7]: add operator keyword

Comment (Saif-Alhaider) [reply client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/usecase/FilterRestaurantsUseCase.kt:7]: and don't return  list of restaurant

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/usecase/FilterRestaurantsUseCase.kt [hunk 2]
```diff
@@ -2,45 +2,17 @@ package org.thechance.common.domain.usecase
...
-        return remoteGateway.getRestaurants().filter {
-            it.priceLevel == priceLevel &&
-                    when {
-                        rating.rem(1) > 0.89 || rating.rem(1) == 0.0 || rating.rem(1) > 0.5
-                        -> it.rating in floor(rating) - 0.1..0.49 + floor(rating)
-
-                        else -> it.rating in 0.5 + floor(rating)..0.89 + floor(rating)
-                    }
-        }
+    override suspend fun invoke(rating: Double, priceLevel: Int): List<Restaurant> {
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/usecase/FilterRestaurantsUseCase.kt:13]: add operator keyword

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/getway/IRemoteGateway.kt [hunk 3]
```diff
@@ -20,4 +20,13 @@ interface IRemoteGateway {
...
 
     suspend fun loginUser(username: String, password: String): UserTokens
+
+    suspend fun filterRestaurants(rating: Double, priceLevel: Int): List<Restaurant>
+
+    suspend fun searchFilterRestaurants(
+        restaurantName: String,
+        rating: Double,
+        priceLevel: Int
+    ): List<Restaurant>
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/getway/IRemoteGateway.kt:30]: don't return list of restaurant

**Review by Saif-Alhaider — APPROVED** [review]

**Review by ay0o5h — APPROVED** [review]

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/gateway/FakeRemoteGateway.kt [hunk 4]
```diff
@@ -279,6 +298,34 @@ class FakeRemoteGateway : IRemoteGateway {
...
+        restaurantName: String,
+        rating: Double,
+        priceLevel: Int
+    ): List<Restaurant> {
+        return searchRestaurantsByRestaurantName(restaurantName).filter {
+            it.priceLevel == priceLevel &&
+                    when {
+                        rating.rem(1) > 0.89 || rating.rem(1) == 0.0 || rating.rem(1) > 0.5
+                        -> it.rating in floor(rating) - 0.1..0.49 + floor(rating)
+
```
Comment (kamelkiko) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/gateway/FakeRemoteGateway.kt:323]: i see you have you repeated this logic , you can put it in a function will be more clean ♥

**Review by kamelkiko — CHANGES_REQUESTED** [review]

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/usecase/SearchFilterRestaurantsUseCase.kt [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package org.thechance.common.domain.usecase
+
+import org.thechance.common.domain.entity.Restaurant
+import org.thechance.common.domain.getway.IRemoteGateway
+
+interface ISearchFilterRestaurantsUseCase {
+    suspend fun invoke(
```
Comment (Karrar-Mohammed) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/usecase/SearchFilterRestaurantsUseCase.kt:7]: add operator keyword

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreen.kt [hunk 6]
```diff
@@ -60,75 +61,100 @@ class RestaurantScreen :
...
+        ) {
+            BpSimpleTextField(
+                modifier = Modifier.widthIn(min = 340.dp, max = 440.dp),
+                hint = "Search for restaurants",
+                onValueChange = listener::onSearchChange,
+                text = state.search,
+                keyboardType = KeyboardType.Text,
+                trailingPainter = painterResource("ic_search.svg")
+            )
+            Column {
```
Comment (Karrar-Mohammed) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreen.kt:99]: you can put this column in separate private function

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt [hunk 7]
```diff
@@ -43,15 +45,15 @@ class RestaurantScreenModel(
 
     private fun filterRestaurants(rating: Double, priceLevel: Int) {
         tryToExecute(
-            { filterRestaurants.filterRestaurants(rating, priceLevel) },
+            { filterRestaurants.invoke(rating, priceLevel) },
```
Comment (Karrar-Mohammed) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt:48]: you can remove .invoke and use the operator keyword on the invoke function

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt [hunk 8]
```diff
@@ -43,15 +45,15 @@ class RestaurantScreenModel(
...
             ::onGetRestaurantSuccessfully,
             ::onError
         )
     }
 
     private fun searchFilterRestaurants(restaurantName: String, rating: Double, priceLevel: Int) {
         tryToExecute(
-            { filterRestaurants.searchFilterRestaurants(restaurantName, rating, priceLevel) },
+            { searchFilterRestaurants.invoke(restaurantName, rating, priceLevel) },
```
Comment (Karrar-Mohammed) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt:56]: same here

**Review by Karrar-Mohammed — APPROVED** [review]

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt (hunk 7 above)
Comment (Asiasama710) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt:48]: why u add invoke?
u can delete it and use name class

**Review by Asiasama710 — APPROVED** [review]

**Review by kamelkiko — APPROVED** [review]

**Pull request merged**